### PR TITLE
(Setup + Test) DB and Server setup + Finally able to setup correct syntax test for db

### PR DESCRIPTION
**Pull request by NgoMike**
https://trello.com/c/9jPTyCrv/48-m-settting-up-db
https://trello.com/c/QqyzsThj/5-s-setting-up-server-express

### CODE REVIEW

File: db/db.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    db.once('open', () => {
+      done();
+    });
+  });
+
+  it('If it gets here, it proves that db is connected', () => {
+    expect(1).toBe(1);
+  })
+});
```
Comment (leungnamson1023) [inline db/db.test.js:21]: nice setting up the db test so far, would be a good idea to just test an insertion of a primitive data type and see if you can query for it next

**Review by leungnamson1023 — COMMENTED** [review]
looks solid

File: server/server.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
 const express = require('express');
+const db = require('../db/mongodb')
```
Comment (tmkuba) [inline server/server.js:2]: missing semicolon

File: db/db.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const mongoose = require('mongoose');
+
+describe('DB Test', () => {
+  beforeAll((done) => {
+    mongoose.connect('mongodb://localhost/data');
```
Comment (tmkuba) [inline db/db.test.js:5]: You may want to consider using the "dotenv" npm package and a .env file to store your MongoDB server address.  That way it isn't hardcoded as a string in your test & in your mongodb.js file (what if you update one but forget to update the other?)

File: db/db.test.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    let db = mongoose.connection;
+
+    db.on('error', (err) => {
```
Comment (tmkuba) [inline db/db.test.js:9]: Wondering if there is a way to have this inside a test()?  It seems a little strange that the actual test is in beforeAll() and getting to test() indicates success...  I don't know much about mongoose testing though

**Review by tmkuba — COMMENTED** [review]
hi Michael, nice job getting it up and running.  The one thing is maybe having more granular commits might be nice

**Comment by tmkuba** [comment]
I wouldn't consider any of my comments to be blocking changes

File: db/db.test.js (hunk 3 above)
Comment (NgoMike) [reply db/db.test.js:5]: thanks Mark, I will definitely look into this!

File: db/db.test.js (hunk 4 above)
Comment (NgoMike) [reply db/db.test.js:9]: I read up to test for Mongodb/Mongoose, we have to use beforeAll to create the connection and once connection is established, it invokes the done function( in my code: line 18-20).

**Pull request merged**